freeipa/src/main/java/com/sequenceiq/freeipa/util/CookieComparator.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.freeipa.util;
+
+import java.io.Serializable;
+import java.util.Comparator;
+import java.util.Date;
+
+import org.apache.http.cookie.Cookie;
+import org.apache.http.cookie.CookieIdentityComparator;
+
+// Sort preceidence:
+//   1. Cookie name (backwards - doesn't matter)
+//   2. Domain name (backwards - doesn't matter)
+//   3. Cookie path (more specific path is listed first) per RFC-6265 section 5.4.2
+//   4. Expiration (no expiratation is listed first, latest expiration is listed next)
+public class CookieComparator implements Serializable, Comparator<Cookie> {
```

Comment:
Don't you have to use it somewhere? Like when you are sorting above, you have to use this comparator, no?

Reply:
The `FreeIpaCookeieStore` is already sorted:
https://github.com/hortonworks/cloudbreak/blob/865c9d05b03a675e0169b5fa3f17efcbd1c5d98d/freeipa/src/main/java/com/sequenceiq/freeipa/util/FreeIpaCookieStore.java#L15
`private final Set<Cookie> cookies = new TreeSet<>(new CookieComparator());`

Reply:
I see that this is already used in https://github.com/hortonworks/cloudbreak/blob/865c9d05b03a675e0169b5fa3f17efcbd1c5d98d/freeipa/src/main/java/com/sequenceiq/freeipa/util/FreeIpaCookieStore.java#L15